packages/language/src/preprocessor/compiler-options-processor.ts

```diff
@@ -11,44 +11,154 @@
+      let srcCompilerResult: CompilerOptionResult | undefined = undefined;
+
+      // process compiler options from the config
+      if (configCompilerOpts) {
+        const configAbstractOptions =
+          parseAbstractCompilerOptions(configCompilerOpts);
+        configCompilerResult = translateCompilerOptions(configAbstractOptions);
+        // adjust all issues from config
+        this.adjustIssuesFromConfig(configCompilerResult.issues, range);
+        if (!range) {
+          // don't report config issues if we don't have a *PROCESS directive to attach them to
+          configCompilerResult.issues = [];
+        }
+      }
+
+      // process compiler options from the *PROCESS directive
+      if (srcCompilerOpts) {
+        const srcAbstractOptions = parseAbstractCompilerOptions(
+          srcCompilerOpts,
+          range
+            ? range.start + CompilerOptionsProcessor.PROCESS_TOKEN_LENGTH
+            : 0,
+        );
+
+        // TODO @montymxb Jun 6th, 2025: Duplicates across both config & src options are not yet accounted for
+        // due to the separate processing of each src
+
+        srcCompilerResult = translateCompilerOptions(srcAbstractOptions);
+      }
+
+      // merge option results
+      const compilerOptionResult: CompilerOptionResult = {
+        options: {
+          ...configCompilerResult?.options,
+          ...srcCompilerResult?.options,
+        },
+        tokens: [
+          ...(configCompilerResult?.tokens || []),
+          ...(srcCompilerResult?.tokens || []),
+        ],
+        issues: [
+          ...(configCompilerResult?.issues || []),
+          ...(srcCompilerResult?.issues || []),
+        ],
+      };
+
+      if (range) {
+        compilerOptionResult.tokens.unshift(range.token);
+      }
+
       return {
-        result: compilerOptionsResult,
+        result: compilerOptionResult,
         text: newText,
       };
     } else {
       return {
         result: undefined,
-        text,
+        text: newText,
       };
     }
   }
 
+  /**
+   * Helper function to adjust issue ranges & messages from the config
+   * Primarily to try and avoid conflating the sourcing of these issues w/ *PROCESS related issues
+   */
+  private adjustIssuesFromConfig(
+    issues: CompilerOptionIssue[],
+    range?: Range & { token: IToken },
+  ): void {
+    for (const issue of issues) {
+      // only retain issues that have a range
+      if (range) {
+        // adjust range & message for issues from config, placing on *PROCESS directive
+        issue.range = range
+          ? {
+              start: range.start,
+              end: range.start + CompilerOptionsProcessor.PROCESS_TOKEN_LENGTH,
+            }
+          : { start: 0, end: 0 };
+        issue.message = `PLI Plugin Config: ${issue.message}`;
+      } else {
+        // report as is, don't adjust
+        console.warn(`PLI Plugin Config: ${issue.message}`, issue.range);
```

Comment:
Suggestion: As it is, this will print a warning for every issue on every keystroke - even if the config hasn't changed. This will pollute the output log **very quickly**. We should print those only when the config has actually changed.